### ISSUE TITLE
Add reference DOCI energies by Klaas for larger systems.

### ISSUE THREAD

**Issue by lelemmen** [GQCG/ci#14]
Using Klaas' DOCI algorithm, we have a reference for CO@DOCI//6-31G.

Here's an example for a possible test we can set up:

```
BOOST_AUTO_TEST_CASE ( DOCI_co_klaas ) {

    // Klaas' reference DOCI energy for CO
    double reference_doci_energy = -112.8392190587;

    // Do a DOCI calculation based on a given FCIDUMP file
    const std::string co_631g_fcidump = "../tests/reference_data/co_631g.FCIDUMP";
    doci::CI_basis doci (co_631g_fcidump);
    doci::DOCI doci_test (&doci);

    // Calculate the total energy as the sum of the lowest energy eigenstate + the internuclear repulsion
    const doci::State& ground = doci_test.getLowestEigenState();
    double test_doci_energy = ground.getEval() + doci.getInternuclear_repulsion();

    BOOST_CHECK(std::abs(test_doci_energy - reference_doci_energy) < 1.0e-06);
}
```

However, this one takes a lot of time since it's essentially diagonalizing the whole DOCI Hamiltonian matrix. 

- [ ] Add this test to the Davidson solver

**Comment by lelemmen** [comment]
We also have some reference data for LiF@DOCI//6-31G.

```
BOOST_AUTO_TEST_CASE ( DOCI_lif_klaas ) {

    // Klaas' reference DOCI energy for LiF
    double reference_lif_energy = -107.0007150075;

    // Do a DOCI calculation based on a given FCIDUMP file
    const std::string lif_631g_fcidump = "../tests/reference_data/lif_631g.FCIDUMP";
    doci::CI_basis doci (lif_631g_fcidump);
    doci::DOCI doci_test (&doci);

    // Calculate the total energy as the sum of the lowest energy eigenstate + the internuclear repulsion
    const doci::State& ground = doci_test.getLowestEigenState();
    double test_doci_energy = ground.getEval() + doci.getInternuclear_repulsion();

    BOOST_CHECK(std::abs(test_doci_energy - reference_lif_energy) < 1.0e-06);
}
```

- [ ] Add this test to the sparse case
- [ ] Add this test to the Davidson solver

**Comment by lelemmen** [comment]
Apparently, this would be even too much for the sparse case.

**Comment by lelemmen** [comment]
These issues have been addressed in PR #23
